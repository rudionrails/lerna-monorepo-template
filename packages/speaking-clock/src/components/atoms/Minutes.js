import React from "react";
import PropTypes from "prop-types";

const Minutes = ({ minutes }) => (
  <div className="font-weight-light text-muted">
    {minutes === 0 ? "o' clock" : `and ${minutes} minutes`}
  </div>
);

Minutes.propTypes = {
  minutes: PropTypes.number.isRequired,
};

export default Minutes;
