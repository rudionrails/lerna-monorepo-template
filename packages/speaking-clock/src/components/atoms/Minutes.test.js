import React from "react";

// module under test
import Minutes from "./Minutes";

const minutes = 13;

test("to render minutes", () => {
  const subject = shallow(<Minutes minutes={minutes} />);
  expect(subject.html()).toContain("and 13 minutes");
});

test("to render when minutes are zero", () => {
  const subject = shallow(<Minutes minutes={0} />);
  expect(subject.html()).toContain("o&#x27; clock");
});

test("to match snapshot", () => {
  const subject = shallow(<Minutes minutes={minutes} />);
  expect(subject).toMatchSnapshot();
});
