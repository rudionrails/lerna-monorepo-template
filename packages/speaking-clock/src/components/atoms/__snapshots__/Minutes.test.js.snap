// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snapshot 1`] = `
<div
  className="font-weight-light text-muted"
>
  and 13 minutes
</div>
`;
